docs/aws/audit/elbmonitoring/rules/alb_desync_mode_check.mdx

```diff
@@ -23,6 +23,211 @@ CBP
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, make sure you have AWS CLI installed and configured with the necessary permissions. You can install it using pip (Python package installer). 
+
+   ```
+   pip install awscli
+   ```
+   Then configure it using the following command and provide your AWS credentials (Access key ID, Secret access key, Default region name, Default output format).
+
+   ```
+   aws configure
+   ```
+
+2. List all the Application Load Balancers: Use the following AWS CLI command to list all the Application Load Balancers (ALBs) in your account.
+
+   ```
+   aws elbv2 describe-load-balancers --query 'LoadBalancers[?Type==`application`].LoadBalancerArn'
+   ```
+
+3. Describe Load Balancer Attributes: For each ALB, describe the load balancer attributes using the following command. Replace 'your-load-balancer-arn' with the ARN of your load balancer.
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn your-load-balancer-arn
+   ```
+
+4. Check Desync Mitigation Mode: In the output of the above command, look for the attribute 'routing.http.desync-mitigation-mode'. If it is not set to 'defensive' or 'strictest', then the ALB is misconfigured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   First, you need to install Boto3. You can do this by running the command `pip install boto3`. After installing Boto3, you need to configure it. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, AWS Secret Access Key, Default region name, and Default output format when prompted.
+
+2. Import necessary libraries and create a session:
+   In your Python script, you need to import the Boto3 library and create a session using your AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. Create an ELB client and retrieve all load balancers:
+   Using the session you created, you can create an ELB client and then use that client to retrieve all of your load balancers.
+
+   ```python
+   elb = session.client('elbv2')
+   load_balancers = elb.describe_load_balancers()
+   ```
+
+4. Check the Desync Mitigation Mode of each load balancer:
+   You can then iterate over each load balancer and check its Desync Mitigation Mode. If the Desync Mitigation Mode is not set, you have detected a misconfiguration.
+
+   ```python
+   for lb in load_balancers['LoadBalancers']:
+       attributes = elb.describe_load_balancer_attributes(LoadBalancerArn=lb['LoadBalancerArn'])
+       for attr in attributes['Attributes']:
+           if attr['Key'] == 'routing.http.desync_mitigation_mode':
+               if attr['Value'] != 'defensive':
+                   print(f"ALB {lb['LoadBalancerName']} has Desync Mitigation Mode not set")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>